internal/jimm/juju/migrationtarget.go

```diff
@@ -60,7 +63,27 @@ func (j *JujuManager) AbortMigration(ctx context.Context, user *openfga.User, mo
 		// as the migration has already been aborted on the target controller.
 		zapctx.Error(ctx, "failed to delete incoming model migration", zap.Error(err), zap.String("modelUUID", modelUUID))
 	}
-	// TODO(SimoneDutto): delete the model from JIMM's state.
+	model := dbmodel.Model{
+		UUID: sql.NullString{
+			String: modelUUID,
+			Valid:  true,
+		},
+	}
+	err = j.Database.GetModel(ctx, &model)
+	if err != nil {
+		if errors.ErrorCode(err) == errors.CodeNotFound {
+			return nil
+		}
+		// Don't return an error if we fail to get the model.
+		zapctx.Error(ctx, "failed to get model for abort migration", zap.Error(err), zap.String("modelUUID", modelUUID))
```

Comment:
Why not? To avoid confusing Juju? Ah I see the reasoning lower down :+1: 
Consider moving the explanation up higher